### PR TITLE
Fix find_package, enable CPack and support clang-cl

### PR DESCRIPTION
**Pull request by ToKiNoBug**
This PR aims to make `kompute` easier to be used as a C++ SDK.

1. Add support to clang-cl
    Before this clang-cl is treated like g++ or clang++, and many `-W` flags are passed to it, but clang-cl is a MSVC-like frontend.

2. Fix an error when kompute is being imported as a cmake package
     `komputeTargets.cmake` is required in a `find_package` call, but this file is not generated previously.

3. Fix a bug to support fmtlib v10
    fmtlib v10 doesn't think `std::array<char,N>` is formattable, so we use `.data()`.
    
4. Fix `KOMPUTE_OPT_INSTALL`. 
    Previously this option doesn't work because `install` calls will always take effect regardless of the value of `KOMPUTE_OPT_INSTALL`. Now no kompute files will be installed if `KOMPUTE_OPT_INSTALL` is set to OFF.

5. Enable CPack to generate many kinds of software packages like deb, rpm, zip, 7z, mac bundle, NSIS, etc.
    Configurations for many other kinds of packages can be added in future.
    To use cpack, run `cpack -G <generator>` in the build directory. For example:
    ```shell
    cd build
    cpack -G TXZ
    cpack -G DEB
    ```
    This will generate a deb package and a tar.xz package.

### CODE REVIEW

**Comment by ToKiNoBug** [comment]
Oh, I see the python build failed. I'm fixing it.

**Comment by ToKiNoBug** [comment]
Besides, I found that gcc13 can't build kompute because built in fmtlib(v8.1.1) has a `possibly dangling reference to a temporary` warning, and this is turned to an error by extra -W compiler flags. 
Either removing extra -W options, or updating fmtlib to v10.0.0 will fix this problem. I don't know which plan is better.

File: src/Manager.cpp [hunk 1]
```diff
@@ -338,7 +338,7 @@ Manager::createDevice(const std::vector<uint32_t>& familyQueueIndices,
 
     KP_LOG_INFO("Using physical device index {} found {}",
                 physicalDeviceIndex,
-                physicalDeviceProperties.deviceName);
+                physicalDeviceProperties.deviceName.data());
```
Comment (axsaucedo) [inline src/Manager.cpp:341]: Why was this change required?

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -86,3 +90,38 @@ endif()
...
+        # Reset INTERFACE_INCLUDE_DIRECTORIES to a generator expression.
+        set_target_properties(${vk_header_target}
+            PROPERTIES
+            INTERFACE_INCLUDE_DIRECTORIES "$<INSTALL_INTERFACE:include>;$<BUILD_INTERFACE:${vk_header_include_dir}>")
+        #message(STATUS "vk_header_include_dir = ${vk_header_include_di r}")
+        install(TARGETS ${vk_header_target}
+            EXPORT komputeTargets)
+    endif ()
+
+    if(KOMPUTE_OPT_BUILD_PYTHON)
```
Comment (axsaucedo) [inline src/CMakeLists.txt:121]: Wondering if indeed the python components are to be installed, as normally they are internal for the python installation through pip only - however I also don't see harm on doing this

**Comment by axsaucedo** [comment]
> Either removing extra -W options, or updating fmtlib to v10.0.0 will fix this problem. I don't know which plan is better.
Do you have more info on this? I also don't have an huge issue with upgrading fmtlib but we woudl also have to test that it works with the rest of the dependencies (ie spdlog), as well as if there are any other issues / changes required to the codebase.

I would be keen to continue with strict -W options as it ensures a clean / maintainable codebase even if often can be a pain intially when introducing

File: src/CMakeLists.txt (hunk 2 above)
Comment (ToKiNoBug) [reply src/CMakeLists.txt:121]: Python components are not to be installed, python libs are found by find_package from external directories, and I don't think we need to install them. It,seems,that we,don't use builtin python, so we don't need to install it.

The install call doesn't install anything about pybind11 either, it just export this empty target into komputeTargets to satisfy the requirement of cmake.

File: src/Manager.cpp (hunk 1 above)
Comment (ToKiNoBug) [reply src/Manager.cpp:341]: It seems that formatter of `std::array<char,N>` is not included, so I take the address instead.
I know strings in std::array may not end with \0, if you think it's kind of unsafe, I can revert this change and include the corrrsponding fmt headers instead.

Comment (ToKiNoBug) [reply src/Manager.cpp:341]: The type of `deviceName` is `vk::ArrayWrapper1D<char, 256>` but not `std::array<char,N>`, fmt v10 doesn't think it is formattable, so a static assertion fails. However, previous versions of fmtlib can pass compilation, it might be caused by a change in the formatting policy.

If we don't take the address of `deviceName`, we will have to implement a formatter manually, and this will tell fmtlib how to format `vk::ArrayWrapper1D<char,N>`

**Comment by ToKiNoBug** [comment]
> I would be keen to continue with strict -W options as it ensures a clean / maintainable codebase even if often can be a pain intially when introducing

We can update the version of builtin fmtlib to v10.1.1. 

Althoug we can't control the version of external fmtlib, we can report a warning/error if users are building kompute with gcc13 or later with fmtlib v9 or earlier.

File: src/CMakeLists.txt (hunk 2 above)
Comment (axsaucedo) [reply src/CMakeLists.txt:121]: Makes sense

File: src/Manager.cpp (hunk 1 above)
Comment (axsaucedo) [reply src/Manager.cpp:341]: Makes sense

**Comment by axsaucedo** [comment]
> We can update the version of builtin fmtlib to v10.1.1.
Yes this seems reasonable.

> Althoug we can't control the version of external fmtlib, we can report a warning/error if users are building kompute with gcc13 or later with fmtlib v9 or earlier.
Absolutely, if we can do a warning without too much complexity on our side that could be helpful, but agree that we can't control what others bring in - we can make it clear that to minimise warnings requires this version - similarly that we don't run tests with other versions

**Comment by ToKiNoBug** [comment]
> > Absolutely, if we can do a warning without too much complexity on our side that could be helpful, but agree that we can't control what others bring in - we can make it clear that to minimise warnings requires this version - similarly that we don't run tests with other versions


I have updated the version of builtin fmt to 10.1.1. But I found it hard to provide a warning about old fmt without complexity. Older fmt doesn't tell its version, may be there's some tricks to get its version, but the code will be too complex.

**Comment by ToKiNoBug** [comment]
I hope to know if there is anything I should do about this PR. Any problem or requirement?

**Review by axsaucedo — APPROVED** [review]
Looks good - thank you for the contribution
